### PR TITLE
Added Response body parse possibility

### PR DESCRIPTION
**Pull request by sigmen**
Closes https://github.com/potok-digital/soap/issues/35

### CODE REVIEW

File: lib/soap/response.ex [hunk 1]
```diff
@@ -7,4 +11,13 @@ defmodule Soap.Response do
...
         }
+
+  alias Soap.Response.Parser
+
+  @doc """
+  Executing with xml response body as string.
+  If body is an empty then parse/1 returns a full parsed response structure as map.
+  """
+  @spec parse_body(String.t()) :: map()
+  def parse_body(body), do: Parser.parse(body)
```
Comment (Nitrino) [inline lib/soap/response.ex]: It is better to call this function just `parse`

File: lib/soap/response.ex [hunk 2]
```diff
@@ -7,4 +11,13 @@ defmodule Soap.Response do
           request_url: String.t(),
           status_code: pos_integer()
         }
+
+  alias Soap.Response.Parser
+
+  @doc """
+  Executing with xml response body as string.
+  If body is an empty then parse/1 returns a full parsed response structure as map.
```
Comment (Nitrino) [inline lib/soap/response.ex]: in this place you use parse/1

Comment (Nitrino) [reply lib/soap/response.ex]: what does it mean a full parsed response structure if the body is empty?

File: lib/soap/response/parser.ex [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    |> parse_element_values
+  end
+
+  @spec parse_element_values(list()) :: any()
+  defp parse_element_values(elements) do
+    if Enum.all?(elements, &is_map/1), do: Enum.reduce(elements, &Map.merge/2), else: extract_value_from_list(elements)
+  end
+
+  @spec extract_value_from_list(list()) :: any()
+  defp extract_value_from_list(elements) do
```
Comment (Nitrino) [inline lib/soap/response/parser.ex]: use the multi-clause functions 
```elixir
defp extract_value_from_list([element]) do: element
defp extract_value_from_list(elements) do: elements
```

File: lib/soap/response/parser.ex [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+  defp parse_elements(elements) when is_list(elements) do
+    elements
+    |> Enum.map(&parse_record/1)
+    |> parse_element_values
+  end
+
+  @spec parse_element_values(list()) :: any()
+  defp parse_element_values(elements) do
+    if Enum.all?(elements, &is_map/1), do: Enum.reduce(elements, &Map.merge/2), else: extract_value_from_list(elements)
```
Comment (Nitrino) [inline lib/soap/response/parser.ex]: I think this line is too complicated, it's better to write it in a multiline form

Comment (Nitrino) [reply lib/soap/response/parser.ex]: What kind of elements do we get in a given place? 
Why do not we process the resulting elements further?

File: test/soap/response/parser_test.exs [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  test "correct response parsed successful" do
+    xml_body = Fixtures.load_xml("send_service/SendMessageResponse.xml")
+    correctly_parsed_response = %{message: "Hello!"}
+
+    assert Parser.parse(xml_body) == correctly_parsed_response
+  end
+
+  test "when response not include response-tag" do
+    fault_xml = Fixtures.load_xml("send_service/SendMessageFault.xml")
+    assert Parser.parse(fault_xml) == %{}
```
Comment (Nitrino) [inline test/soap/response/parser_test.exs]: why are not you parsing text errors? this is very important part of the answer

File: lib/soap/response.ex (hunk 1 above)
Comment (sigmen) [reply lib/soap/response.ex]: Done

File: lib/soap/response.ex (hunk 2 above)
Comment (sigmen) [reply lib/soap/response.ex]: Done

File: lib/soap/response/parser.ex (hunk 3 above)
Comment (sigmen) [reply lib/soap/response/parser.ex]: Done

**Comment by Nitrino** [comment]
❤️ 💚 💙 💛 💜

**Pull request merged**